### PR TITLE
Fix unit-test inconsistency for promises

### PR DESCRIPTION
**Pull request by tkeri**
Use a helper function to check the availability of promise.

JerryScript-DCO-1.0-Signed-off-by: Tamas Keri tkeri@inf.u-szeged.hu

### CODE REVIEW

File: tests/unit-core/test-promise.c [hunk 1]
```diff
@@ -114,61 +113,79 @@ register_js_function (const char *name_p, /**< name of the function */
...
-  register_js_function ("create_promise1", create_promise1_handler);
-  register_js_function ("create_promise2", create_promise2_handler);
-  register_js_function ("assert", assert_handler);
-
-  jerry_value_t parsed_code_val = jerry_parse ((jerry_char_t *) test_source, strlen (test_source), false);
-  TEST_ASSERT (!jerry_value_has_error_flag (parsed_code_val));
-
-  jerry_value_t res = jerry_run (parsed_code_val);
-  TEST_ASSERT (!jerry_value_has_error_flag (res));
+  if (promise_is_available ())
```
Comment (dbatyai) [inline tests/unit-core/test-promise.c]: I'd prefer an early return here when promises are not available. That way we don't have to reindent the entire file.

File: tests/unit-core/test-promise.c [hunk 2]
```diff
@@ -114,61 +113,79 @@ register_js_function (const char *name_p, /**< name of the function */
...
-  jerry_release_value (res);
-  jerry_release_value (parsed_code_val);
+    jerry_value_t parsed_code_val = jerry_parse ((jerry_char_t *) test_source, strlen (test_source), false);
+    TEST_ASSERT (!jerry_value_has_error_flag (parsed_code_val));
 
-  /* Test jerry_create_promise and jerry_value_is_promise. */
-  TEST_ASSERT (jerry_value_is_promise (my_promise1));
-  TEST_ASSERT (jerry_value_is_promise (my_promise2));
+    jerry_value_t res = jerry_run (parsed_code_val);
+    TEST_ASSERT (jerry_value_has_error_flag (res));
```
Comment (dbatyai) [inline tests/unit-core/test-promise.c]: It seems an exclamation mark is missing from here, but was there in the original line.

File: tests/unit-core/test-promise.c (hunk 1 above)
Comment (tkeri) [reply tests/unit-core/test-promise.c]: Done.

File: tests/unit-core/test-promise.c (hunk 2 above)
Comment (tkeri) [reply tests/unit-core/test-promise.c]: Done.

File: tests/unit-core/test-promise.c [hunk 3]
```diff
@@ -114,11 +113,37 @@ register_js_function (const char *name_p, /**< name of the function */
...
+
+  return has_prop;
+} /* promise_is_available */
+
 int
 main (void)
 {
   jerry_init (JERRY_INIT_EMPTY);
 
+  if (!(promise_is_available ()))
```
Comment (galpeter) [inline tests/unit-core/test-promise.c]: no need for the extra `()` just: `if (!promise_is...())`

File: tests/unit-core/test-promise.c [hunk 4]
```diff
@@ -114,11 +113,37 @@ register_js_function (const char *name_p, /**< name of the function */
...
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Promise is disable!\n");
```
Comment (galpeter) [inline tests/unit-core/test-promise.c]: typo: "disable" -> "disabled"

**Comment by galpeter** [comment]
lgtm

**Pull request merged**